synapse/config/server.py

```diff
@@ -490,6 +500,20 @@ def default_config(self, server_name, data_dir_path, **kwargs):
 
         # Used by phonehome stats to group together related servers.
         #server_context: context
+
+        # Using SQLite with Synapse induces a severe degradation in performances
+        # compared to using PostgreSQL. SQLite should only be used for personal
+        # testing. Production instances should use PostgreSQL instead. You can
+        # find documentation on setting up PostgreSQL with Synapse here:
+        # https://github.com/matrix-org/synapse/blob/master/docs/postgres.rst
+        #
+        # If you want to use federation with SQLite regardless, you can
+        # uncomment the line below, but be aware that it can make Synapse
+        # malfunction and be very laggy, especially when joining large rooms.
+        # This option defaults to 'false' when using SQLite and 'true'
+        # otherwise.
+        #
+        #enable_federation_can_cause_bad_perfs_with_sqlite = true
```

Comment:
I'm finding this name a bit misleading, tbh. The "with_sqlite" makes me think that it will be ignored with postgres.

I suggest one of two actions:
* rename the setting to just `enable_federation`, and keep its existing behaviour where it defaults to `true` for postgres and `false` for sqlite
* keep the existing name (or possibly `enable_federation_on_sqlite_even_though_it_is_slow`), and *ignore* it when we are using postgres.

of those, option 2 sounds better?

Reply:
For context at first I named it just `enable_federation`, but @aaronraimist [argued](https://github.com/matrix-org/synapse/pull/5078#pullrequestreview-228388884) that:

> I feel like a lot of people don't read the comments, or they use some package that doesn't provide the full comment above the command, just a huge list of commands.

Which is why I renamed it.

Option 2 sounds better, I'm going with it.